rpc, docs: Add note for commands that supports only legacy wallets

yusufsahinhamza opened the pull request:
Refs #25363, apparently issue is not updated since over a month, so i decided to put the same `importaddress` note in #25368 to other rpc commands that needs this note.

Note is added for following commands:

- `importprivkey`
- `importpubkey`
- `importwallet`
- `dumpprivkey`
- `dumpwallet`
- `importmulti`
- `addmultisigaddress`
- `sethdseed`

jonatack reviewed (commented):
This note you have copied over from #25368 does not seem relevant to these RPCs. In some cases, the error message may need to be updated as well.

yusufsahinhamza commented:
@jonatack 

Looks like no need of this note for `getnewaddress`.

According to this release note:

https://github.com/bitcoin/bitcoin/blob/194f6dc43ccc330a8a4607be3a2b8935490d6db0/doc/release-notes/release-notes-0.21.0.md?plain=1#L412-L422

We can simply add **"Note: This command is only compatible with legacy wallets."** note to these RPCs, but for example where there is alternative command to use for descriptor wallets; e.g. release note says:

https://github.com/bitcoin/bitcoin/blob/194f6dc43ccc330a8a4607be3a2b8935490d6db0/doc/release-notes/release-notes-0.21.0.md?plain=1#L405-L406

In this case for `importmulti` we can add additional note **"Use "importdescriptors" with addr(X)" for descriptor wallets."** after the first note to point out what to use as an alternative.

luke-jr commented:
>getnewaddress

?!?!?!

yusufsahinhamza commented:
> >getnewaddress
> 
> 
> 
> ?!?!?!

Didn't do any changes as i'm waiting for an answer about my last message, or let's say suggestion to what to do.

yusufsahinhamza committed:
rpc, docs: Add note for commands that supports only legacy wallets

Note is added for following rpc commands:

importprivkey, importpubkey, importwallet, dumpprivkey,
dumpwallet, importmulti, addmultisigaddress, sethdseed

achow101 commented on src/wallet/rpc/backup.cpp:
```diff
@@ -1258,7 +1263,8 @@ RPCHelpMan importmulti()
             "but the key was used to create transactions, rescanwallet needs to be called with the appropriate block range.\n"
-            "Note: Use \"getwalletinfo\" to query the scanning progress.\n",
+            "Note: Use \"getwalletinfo\" to query the scanning progress.\n"
+            "Note: This command is only compatible with legacy wallets. Use \"importdescriptors\" with \"addr(X)\" for descriptor wallets.\n",
```
I don't think it's useful to mention `addr()` here as `importmulti` allows the import of a ton of different things, including descriptors. Mentioning a specific descriptor type, which only applies to one of the things that `importmulti` can deal with, doesn't seem to make sense to me.

yusufsahinhamza replied:
@achow101 I agree, I'd rather like it as **"Use "importdescriptors" for descriptor wallets."**, mentioning a specific parameter doesn't seem necessary. People would get help by using only **"importdescriptors"** anyway.

This line can also be changed:
https://github.com/bitcoin/bitcoin/blob/1abbae65eb9c99df5d8941008068d83ad99bf117/src/wallet/rpc/backup.cpp#L213

tusher456 commented:
32sMEMSwnLLcuxzHptTDksk9auSPoccWiJ

On Fri, 26 Aug 2022, 5:07 am DrahtBot, ***@***.***> wrote:

> 🐙 This pull request conflicts with the target branch and needs rebase
> <https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#rebasing-changes>
> .
>
> Want to unsubscribe from rebase notifications on this pull request? Just
> convert this pull request to a "draft".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/25680#issuecomment-1227839558>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2RJYJW7TY42CRWRBVJBUB3V274EDANCNFSM54NFLULA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

yusufsahinhamza commented:
775ba9ccea0adcae349787647e8aa08844ff390d Conflicts are resolved and rebased.

theStack reviewed (commented):
Concept ACK

For the `importprivkey` and `importpubkey` RPC notes, it could make sense to suggest using `importdescriptor` with `combo(X)` for descriptor wallets?

yusufsahinhamza commented:
> For the `importprivkey` and `importpubkey` RPC notes, it could make sense to suggest using `importdescriptor` with `combo(X)` for descriptor wallets?

@theStack It seems to make sense for me. I can rebase and push changes if no other suggestion about that.

achow101 commented:
ACK 9141e4395a5f923059ad61ac6ba42a1a89e92cb0

luke-jr commented:
@achow101 Why was this merged?

I'm not sure suggesting `combo()` is a good idea. Isn't combo only for having a migration path, not something we want people to actually use?

achow101 commented:
While I don't think we should necessarily encourage people to use `combo()`, I don't think it's bad to mention it either. in the case of `importpubkey` and `importprivkey`, `combo()` with `importdescriptors` does replicate their behavior so it isn't incorrect. I don't really care either way whether it's mentioned or not.
